Source/WebKit/UIProcess/WebFrameProxy.cpp

```diff
@@ -349,15 +349,10 @@ void WebFrameProxy::disconnect()
 
 void WebFrameProxy::didCreateSubframe(WebCore::FrameIdentifier frameID)
 {
-    // The DecidePolicyForNavigationActionSync IPC is synchronous and may therefore get processed before the DidCreateSubframe one.
-    // When this happens, decidePolicyForNavigationActionSync() calls didCreateSubframe() and we need to ignore the DidCreateSubframe
-    // IPC when it later gets processed.
-    if (WebFrameProxy::webFrame(frameID))
-        return;
-
-    MESSAGE_CHECK(m_process, m_page);
     MESSAGE_CHECK(m_process, WebFrameProxy::canCreateFrame(frameID));
     MESSAGE_CHECK(m_process, frameID.processIdentifier() == m_process->coreProcessIdentifier());
+    if (!m_page)
```

Comment:
We're replacing an assertion with a null check, always concerning. Why can this happen?

Reply:
I recently added the MESSAGE_CHECK(m_process, m_page) and I think it's incorrect.  If we receive a message to the frame after the frame has disconnected, it doesn't indicate that the process has been compromised and needs to be terminated.  It indicates that messages were received in a rare order and this one should just be ignored.